Extensions/Guardrail+Settings.swift

```diff
@@ -9,8 +9,76 @@
 import HealthKit
 
 public extension Guardrail where Value == HKQuantity {
-    static let correctionRange = Guardrail(absoluteBounds: 60...180, recommendedBounds: 70...120, unit: .milligramsPerDeciliter)
+    static let suspendThreshold = Guardrail(absoluteBounds: 67...110, recommendedBounds: 74...80, unit: .milligramsPerDeciliter)
 
+    static func maxSuspendThresholdValue(correctionRangeSchedule: GlucoseRangeSchedule?, preMealTargetRange: DoubleRange?, workoutTargetRange: DoubleRange?, unit: HKUnit) -> HKQuantity {
+
+        return [
+            suspendThreshold.absoluteBounds.upperBound.doubleValue(for: unit),
+            correctionRangeSchedule?.minLowerBound().doubleValue(for: unit),
+            preMealTargetRange?.minValue,
+            workoutTargetRange?.minValue
+        ]
+        .compactMap { $0 }
+        .min()
+        .map { HKQuantity(unit: unit, doubleValue: $0) }!
+    }
+
+    static let correctionRange = Guardrail(absoluteBounds: 87...180, recommendedBounds: 101...115, unit: .milligramsPerDeciliter)
+
+    static func minCorrectionRangeValue(suspendThreshold: GlucoseThreshold?, unit: HKUnit) -> HKQuantity {
+        return [
+            correctionRange.absoluteBounds.lowerBound.doubleValue(for: unit),
+            suspendThreshold?.value
+        ]
+        .compactMap { $0 }
+        .max()
+        .map { HKQuantity(unit: unit, doubleValue: $0) }!
+    }
+    
+    fileprivate static func workoutCorrectionRange(correctionRangeScheduleRange: ClosedRange<HKQuantity>,
+                                                   suspendThreshold: GlucoseThreshold?,
+                                                   unit: HKUnit) -> Guardrail<HKQuantity> {
+        // Static "unconstrained" constant values before applying constraints
+        let workoutCorrectionRange = Guardrail(absoluteBounds: 85...250, recommendedBounds: 101...180, unit: .milligramsPerDeciliter)
+        
+        let absoluteLowerBound = [
+            workoutCorrectionRange.absoluteBounds.lowerBound.doubleValue(for: unit),
+            suspendThreshold?.value
+        ]
+        .compactMap { $0 }
+        .max()
+        .map { HKQuantity(unit: unit, doubleValue: $0) }!
+        let recommmendedLowerBound = max(absoluteLowerBound, correctionRangeScheduleRange.upperBound)
+        return Guardrail(
+            absoluteBounds: absoluteLowerBound...workoutCorrectionRange.absoluteBounds.upperBound,
+            recommendedBounds: recommmendedLowerBound...workoutCorrectionRange.recommendedBounds.upperBound
+        )
+    }
+    
+    fileprivate static func preMealCorrectionRange(correctionRangeScheduleRange: ClosedRange<HKQuantity>,
+                                                   suspendThreshold: GlucoseThreshold?,
+                                                   unit: HKUnit) -> Guardrail<HKQuantity> {
+        let premealCorrectionRangeMaximum = HKQuantity(unit: unit, doubleValue: 130.0)
```

Comment:
again there is an assumption made on the units here. I'm guessing you don't mean 130 mmol/L